### PR TITLE
Added logs and emit iframe event for video and audio sharing

### PR DESCRIPTION
**Pull request by apetrus20**

### CODE REVIEW

File: modules/API/API.js [hunk 1]
```diff
@@ -1770,6 +1770,24 @@ class API {
...
+     *
+     * @param {string} mediaType - Whether the content which is being shared is audio or video.
+     * @param {string} value - Whether the sharing is playing, pause or stop (on audio there is only playing and stop).
+     * @param {string} participantId - Participant id of the participant which started or ended
+     *  the video or audio sharing.
+     * @returns {void}
+     */
+    notifyAudioOrVideoSharingToggled(mediaType, value, participantId) {
+        this._sendEvent({
+            name: 'audio-or-video-sharing-toggled',
```
Comment (robertpin) [inline modules/API/API.js:1784]: Considering these are two separate features, shouldn't we have two separate events? ()one for audio and another for video

**Comment by robertpin** [comment]
Please remove empty file react/features/screen-share/constants.ts

File: react/features/screen-share/middleware.js [hunk 2]
```diff
@@ -3,9 +3,12 @@
 import { CONFERENCE_JOINED } from '../base/conference';
 import { MiddlewareRegistry } from '../base/redux';
 
-import { SET_SCREENSHARE_CAPTURE_FRAME_RATE } from './actionTypes';
+import { SET_SCREENSHARE_CAPTURE_FRAME_RATE, SET_SCREEN_AUDIO_SHARE_STATE } from './actionTypes';
+import AUDIO from './constants';
```
Comment (robertpin) [inline react/features/screen-share/middleware.js:7]: The constants file is empty. Also, can't we use `MEDIA_TYPE.AUDIO`?

File: react/features/screen-share/middleware.js [hunk 3]
```diff
@@ -26,6 +31,19 @@ MiddlewareRegistry.register(store => next => action => {
...
+
+    case SET_SCREEN_AUDIO_SHARE_STATE: {
+        const { isSharingAudio } = action;
+        const { participantId } = state['features/large-video'];
+
+        if (isSharingAudio) {
+            logger.debug(`User with id: ${participantId} playing audio sharing.`);
+            APP.API.notifyAudioOrVideoSharingToggled(AUDIO, 'playing', participantId);
+        } else {
+            logger.debug(`User with id: ${participantId} stop video sharing.`);
```
Comment (robertpin) [inline react/features/screen-share/middleware.js:43]: `stop audio`?

File: react/features/shared-video/constants.js [hunk 4]
```diff
@@ -22,6 +22,13 @@ export const YOUTUBE_PLAYER_PARTICIPANT_NAME = 'YouTube';
  */
 export const SHARED_VIDEO = 'shared-video';
 
+/**
+ * Shared video type.
+ *
+ * @type {string}
+ */
+export const VIDEO = 'video';
```
Comment (robertpin) [inline react/features/shared-video/constants.js:30]: Do we need this? Can't we use `MEDIA_TYPE.VIDEO` ?

**Review by robertpin — CHANGES_REQUESTED** [review]

**Comment by robertpin** [comment]
Please do not merge master. Rebase instead

File: react/features/shared-video/constants.js (hunk 4 above)
Comment (apetrus20) [reply react/features/shared-video/constants.js]: done

File: react/features/screen-share/middleware.js (hunk 2 above)
Comment (apetrus20) [reply react/features/screen-share/middleware.js]: done

File: modules/API/API.js (hunk 1 above)
Comment (apetrus20) [reply modules/API/API.js:1784]: This was agreed with Saul when I started the implementation, only one event with different type for audio/video.

File: react/features/screen-share/middleware.js [hunk 5]
```diff
@@ -26,6 +31,19 @@ MiddlewareRegistry.register(store => next => action => {
         _setScreenshareCaptureFps(store, captureFrameRate);
         break;
     }
+
+    case SET_SCREEN_AUDIO_SHARE_STATE: {
+        const { isSharingAudio } = action;
+        const { participantId } = state['features/large-video'];
```
Comment (robertpin) [inline react/features/screen-share/middleware.js:37]: Why do we take the `participantId` from `large-video`? How is audio sharing related to large video?

**Pull request closed without merging**